aicsimageio/aics_image.py

```diff
@@ -186,11 +186,20 @@ def determine_reader(
                                 fs_kwargs=fs_kwargs,
                             ):
                                 return ReaderClass
+
                         except Exception as e:
-                            log.warning(
-                                f"Attempted file ({path}) load with "
-                                f"reader: {reader} failed with error: {e}"
-                            )
+                            if str(e) == "No module named 'bfio'":
+                                log.warning(
```

Comment:
I wonder if this should be an info log

Reply:
Agreed, reducing severity might help our users who just have no desire to install bfio and don't want to see the warning all the time.

Also, doing a string comparison is brittle here:  if we ever change the message in the place where it originates, how would a programmer know to come here and change this string comparison?  A possible improvement would be to make a new exception type and catch the more specific type.

Reply:
a new exception sounds good. I assume for this package I would make that in the exceptions.py file? also, what would the implementation of this error look like? would it just be wrapping the import [here ](https://github.com/AllenCellModeling/aicsimageio/blob/6dac4f34ffe410474d14d6337197852b92a7d83c/aicsimageio/readers/bfio_reader.py#L9)?

Reply:
maybe something like this:
```
except ModuleNotFoundError as error1:
    # check for error1.name == bfio and do stuff
except Exception as e:
    # respond to all other exceptions...
```

Reply:
also consider not even checking for bfio - any ModuleNotFoundError is pretty much guaranteed to be a failed import, and usually would be because some dependency is not installed.    Is the exception coming from _load_reader or is_supported_image?   Maybe could even handle ModuleNotFoundErrors from _load_reader separately and certainly with a different log message.

Reply:
> 

I think this would be an issue for other Modules that might not be installed that we actually do want to report. would something like this work
```                       
 except Exception as e:
      if e.__class__ == ModuleNotFoundError and ("'bfio'") in str(e):
          log.warning(
              f"Attempted file ({path}) "
              f"load with reader: {reader}"
              f"for better performance, "
              f"consider installing 'bfio'."
          )
      else:
          log.warning(
              f"Attempted file ({path}) load with "
              f"reader: {reader} failed with error: {e}"
          )

Reply:
> also consider not even checking for bfio - any ModuleNotFoundError is pretty much guaranteed to be a failed import, and usually would be because some dependency is not installed. Is the exception coming from _load_reader or is_supported_image? Maybe could even handle ModuleNotFoundErrors from _load_reader separately and certainly with a different log message.

The error comes from `ReaderClass = _load_reader(reader)`.

Reply:
the check `if e.__class__` is built into this:
```
try:
    # call _load_reader etc
except ModuleNotFoundError as e:
    # handle ModuleNotFoundError
except Exception as e:
    # handle all other exceptions
```
is doing.
And to get even more specific I was saying you could do a try/except around each line of code separately.  So _load_reader would be expected to raise a certain type of exception (like a failed import) that maybe is_supported_image would not.

Reply:
(The way we auto-select which Reader to use is a big challenge and will come up again when we start talking about the refactor)